Mathlib/Analysis/InnerProductSpace/Simultaneous.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jack Cheverton, Samyak Dhar Tuladhar
+-/
+
+import Mathlib.Analysis.InnerProductSpace.Spectrum
+import Mathlib.Analysis.InnerProductSpace.Projection
+import Mathlib.Order.CompleteLattice
+
+/-! # Simultaneous eigenspaces of a commuting pair of symmetric operators
+
+This file collects various decomposition results for simultaneous eigenspaces of a commuting pair
+of symmetric operators on a finite-dimensional Hilbert space.
+
+# Main Result
+
+* `DirectSum.IsInternal_of_simultaneous_eigenspaces_of_commuting_symmetric_pair` establishes that
+   if `{A B : E →ₗ[𝕜] E}`, then `IsSymmetric A`, `IsSymmetric B` and `A ∘ₗ B = B ∘ₗ A` imply that
+   `E` decomposes as an internal direct sum of the pairwise orthogonal spaces
+   `eigenspace B μ ⊓ eigenspace A ν`
+
+## TODO
+
+Develop `Diagonalization` class that selects a basis of (simultaneous) eigenvectors in a
+principled way.
+
+## Tags
+
+self-adjoint operator, simultaneous eigenspaces, simultaneous diagonalization
+
+-/
+
+variable {𝕜 : Type*} [RCLike 𝕜]
+variable {E : Type*} [NormedAddCommGroup E] [InnerProductSpace 𝕜 E] [FiniteDimensional 𝕜 E]
+
+local notation "⟪" x ", " y "⟫" => @inner 𝕜 E _ x y
+
+open Module.End
+
+namespace LinearMap
+
+namespace IsSymmetric
+
+section Pair
+
+variable {α : 𝕜} {A B : E →ₗ[𝕜] E} (hA : A.IsSymmetric) (hB : B.IsSymmetric)
+    (hAB : A ∘ₗ B = B ∘ₗ A)
```

Comment:
This should be defeq, but I'm not sure it's actually better, so I'm not necessarily advocating for it yet. Mainly I'm putting this here as a reminder to myself that it's something to consider.
```suggestion
    (hAB : Commute A B)
```

Reply:
Yes...somehow this doesn't work. I was wondering why that was when we originally wrote this, because checking...these are, I think, definitionally equal.